src/backend/crud/agent.py

```diff
@@ -74,16 +84,26 @@ def get_agents(
     """
     query = db.query(Agent)
 
+    if visibility == AgentVisibility.PUBLIC:
+        # Public agents -> is_private = False or None
+        query = query.filter((Agent.is_private == False) | (Agent.is_private == None))
```

Comment:
With the default, we can avoid making this nullable now - and remove the Agent.is_private == None logic here